pkg/client/build.go

```diff
@@ -1042,6 +1043,30 @@ func (c *Client) fetchBuildpack(ctx context.Context, bp string, relativeBaseDir
 		fetchedBPs = append(append(fetchedBPs, mainBP), depBPs...)
 		mainBPInfo := mainBP.Descriptor().Info()
 		moduleInfo = &mainBPInfo
+
+		packageCfgPath := filepath.Join(bp, "package.toml")
+		_, err = os.Stat(packageCfgPath)
+		if err == nil {
+			packageReader := buildpackage.NewConfigReader()
```

Comment:
As a small readability improvement we can extract this part into a small method, having something like this:

```
if err == nil {
			fetchedDeps, err := c.fetchBPsDependencies(ctx, packageCfgPath, registry, imageOS, bp, publish, pullPolicy)
			if err != nil {
				//handle the error
			}
			fetchedBPs = append(fetchedBPs, fetchedDeps...)
		}
	}
	return fetchedBPs, moduleInfo, nil
```

with an extracted function like:
 ```
 func (c Client) fetchBPsDependencies(ctx context.Context, packageCfgPath, registry, imageOS, bp string, publish bool, pullPolicy image.PullPolicy) ([]buildpack.BuildModule, error) {
	packageReader := buildpackage.NewConfigReader()
	packageCfg, err := packageReader.Read(packageCfgPath)
	if err == nil {
		fetchedBPs := []buildpack.BuildModule{}
		for _, dep := range packageCfg.Dependencies {
			mainBP, deps, err := c.buildpackDownloader.Download(ctx, dep.URI, buildpack.DownloadOptions{
				RegistryName:    registry,
				ImageOS:         imageOS,
				RelativeBaseDir: filepath.Join(bp, packageCfg.Buildpack.URI),
				Daemon:          !publish,
				PullPolicy:      pullPolicy,
			})

			if err != nil {
				return nil, errors.Wrapf(err, "fetching dependencies (uri=%s,image=%s)", style.Symbol(dep.URI), style.Symbol(dep.ImageName))
			}
			fetchedBPs = append(append(fetchedBPs, mainBP), deps...)
		}
		return fetchedBPs, nil
	}
	return nil, err
}
```

It will improve the readability, keep the function short, making it easier applying future improvements

Reply:
Sounds good to me... to avoid passing all those arguments to `fetchBPsDependencies` you could just pass the original `downloadOptions` and mutate the `RelativeBaseDir`